### PR TITLE
[FLINK-26270] Flink SQL write data to kafka by CSV format , whether d…

### PR DESCRIPTION
**Pull request by sunshineJK**
…ecimal type was converted to scientific notation

## What is the purpose of the change

To solve csv serialization , avoiding decimal conversion to scientific notation

## Brief change log

- CsvFormatOptions add write-bigdecimal-as-plain, Optional whether write bigdecimal using scientific notation
- csvMapper.configure(JsonGenerator.Feature.WRITE_BIGDECIMAL_AS_PLAIN, plain)

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (yes)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: docs/content.zh/docs/connectors/table/formats/csv.md [hunk 1]
```diff
@@ -29,18 +29,17 @@ under the License.
 {{< label "Format: Serialization Schema" >}}
 {{< label "Format: Deserialization Schema" >}}
 
-[CSV](https://zh.wikipedia.org/wiki/%E9%80%97%E5%8F%B7%E5%88%86%E9%9A%94%E5%80%BC) Format 允许我们基于 CSV schema 进行解析和生成 CSV 数据。 目前 CSV schema 是基于 table schema 推断而来的。
```
Comment (MartijnVisser) [inline docs/content.zh/docs/connectors/table/formats/csv.md:32]: You have updated the Chinese documentation with English content. The English content can be found in `docs/content/docs/connectors/table/formats/csv.md` - Please make sure that at least the English one has the right documentation

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvCommons.java [hunk 2]
```diff
@@ -35,6 +35,7 @@
 import static org.apache.flink.formats.csv.CsvFormatOptions.IGNORE_PARSE_ERRORS;
 import static org.apache.flink.formats.csv.CsvFormatOptions.NULL_LITERAL;
 import static org.apache.flink.formats.csv.CsvFormatOptions.QUOTE_CHARACTER;
+import static org.apache.flink.formats.csv.CsvFormatOptions.WRITE_BIGDECIMAL_AS_PLAIN;
```
Comment (MartijnVisser) [inline flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvCommons.java:38]: @afedulov or @PatrickRen any opinions about this solution implementation?

**Review by MartijnVisser — CHANGES_REQUESTED** [review]
@sunshineJK Thanks for the PR, but there are some changes that need to be incorporated. Also, I'm missing a updated/new test for this functionality.

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvCommons.java (hunk 2 above)
Comment (PatrickRen) [reply flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvCommons.java:38]: I'm not quite familiar with Jackson but I think we need a test at least. @sunshineJK Could you add a test case to validate this feature?

Comment (sunshineJK) [reply flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvCommons.java:38]: Okay, I'll add that test case later

File: docs/content.zh/docs/connectors/table/formats/csv.md (hunk 1 above)
Comment (sunshineJK) [reply docs/content.zh/docs/connectors/table/formats/csv.md]: Has been updated

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvCommons.java (hunk 2 above)
Comment (sunshineJK) [reply flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvCommons.java:38]: hello，I submitted the test case

File: docs/content.zh/docs/connectors/table/formats/csv.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ title: CSV
 weight: 2
 type: docs
 aliases:
-  - /zh/dev/table/connectors/formats/csv.html
```
Comment (afedulov) [inline docs/content.zh/docs/connectors/table/formats/csv.md:6]: Is this change necessary?

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java [hunk 4]
```diff
@@ -87,5 +87,11 @@ public class CsvFormatOptions {
                             "Optional null literal string that is interpreted as a\n"
                                     + "null value (disabled by default)");
 
+    public static final ConfigOption<Boolean> WRITE_BIGDECIMAL_AS_PLAIN =
+            ConfigOptions.key("write-bigdecimal-as-plain")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription("Optional whether write bigdecimal using scientific notation");
```
Comment (afedulov) [inline flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java:94]: This needs a better comment since the relationship between the flag values and the scientific notation is not immediately clear to the user of the API. The comment has to explain what is the outcome if this flag is set to true.

**Review by afedulov — CHANGES_REQUESTED** [review]
Thanks for the contribution @sunshineJK. It generally looks good, I left a couple of minor comments.

File: docs/content.zh/docs/connectors/table/formats/csv.md (hunk 3 above)
Comment (sunshineJK) [reply docs/content.zh/docs/connectors/table/formats/csv.md:6]: sorry, I will revert

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java (hunk 4 above)
Comment (sunshineJK) [reply flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java]: Thank you for your advice,I've made some updates

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java [hunk 5]
```diff
@@ -87,5 +87,11 @@ public class CsvFormatOptions {
                             "Optional null literal string that is interpreted as a\n"
                                     + "null value (disabled by default)");
 
+    public static final ConfigOption<Boolean> WRITE_BIGDECIMAL_AS_PLAIN =
+            ConfigOptions.key("write-bigdecimal-as-plain")
+                    .booleanType()
+                    .defaultValue(true)
```
[the remark this answers is not in the document]
Comment (sunshineJK) [reply flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java]: The current default value is equal to "standard" Jackson

[the remark this answers is not in the document]
Comment (afedulov) [reply flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java]: Are you sure?
https://github.com/FasterXML/jackson-core/blob/5a8ed0824491a1910aeea4ca7164150f13e4e446/src/main/java/com/fasterxml/jackson/core/JsonGenerator.java#L201

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java [hunk 6]
```diff
@@ -87,5 +87,12 @@ public class CsvFormatOptions {
...
                                     + "null value (disabled by default)");
 
+    public static final ConfigOption<Boolean> DISABLE_WRITE_BIGDECIMAL_AS_SCIENTIFIC_NOTATION =
+            ConfigOptions.key("disable-write-bigdecimal-as-scientific-notation")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Disabled representation of data of type Bigdecimal as scientific notation (default is false, which requires conversion to scientific notation), e.g. A Bigdecimal number of 100000, If false the result is '1E+5', if true the result is 100000.");
+
     private CsvFormatOptions() {}
```
Comment (sunshineJK) [inline flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java:97]: hi，@afedulov I've changed this parameter and now the default is false

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java [hunk 7]
```diff
@@ -87,11 +87,12 @@ public class CsvFormatOptions {
                             "Optional null literal string that is interpreted as a\n"
                                     + "null value (disabled by default)");
 
-    public static final ConfigOption<Boolean> WRITE_BIGDECIMAL_AS_PLAIN =
-            ConfigOptions.key("write-bigdecimal-as-plain")
+    public static final ConfigOption<Boolean> DISABLE_WRITE_BIGDECIMAL_AS_SCIENTIFIC_NOTATION =
```
Comment (afedulov) [inline flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java:90]: Why not just `WRITE_BIGDECIMAL_IN_SCIENTIFIC_NOTATION` without DISABLE? The boolean already incorporates this notion. (also mind IN instead of AS)

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java [hunk 8]
```diff
@@ -87,11 +87,12 @@ public class CsvFormatOptions {
...
+            ConfigOptions.key("disable-write-bigdecimal-as-scientific-notation")
                     .booleanType()
-                    .defaultValue(true)
-                    .withDescription("Optional whether write bigdecimal using scientific notation");
+                    .defaultValue(false)
+                    .withDescription(
+                            "Disabled representation of data of type Bigdecimal as scientific notation (default is false, which requires conversion to scientific notation), e.g. A Bigdecimal number of 100000, If false the result is '1E+5', if true the result is 100000.");
```
Comment (afedulov) [inline flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFormatOptions.java:95]: `DisableS`
as -> in

**Comment by afedulov** [comment]
Hi @sunshineJK, I can take a look on Friday, sorry, a very busy week.

**Comment by sunshineJK** [comment]
> Hi @sunshineJK, I can take a look on Friday, sorry, a very busy week.

OK,thank you

**Review by afedulov — CHANGES_REQUESTED** [review]
Hi @sunshineJK looks mostly good, just a couple of nits from my side.

**Comment by sunshineJK** [comment]
hi @afedulov Please review it again

**Comment by sunshineJK** [comment]
> 
Thank you , I'm going to modify it a little bit.

**Review by afedulov — APPROVED** [review]
Thanks, @sunshineJK. LGTM.

**Comment by sunshineJK** [comment]
> Thanks, @sunshineJK. LGTM.
@afedulov Thank you for your advice. I have gained a lot.

**Comment by sunshineJK** [comment]
Hi @MartijnVisser , What should I do next

**Comment by sunshineJK** [comment]
Hi @afedulov @MartijnVisser @PatrickRen What are the next steps? Do I need to do any more operations

**Review by MartijnVisser — APPROVED** [review]
@sunshineJK LGTM % CI passes. I've rebased your PR to make sure that it still builds properly

**Review by sunshineJK — COMMENTED** [review]
Thanks,@MartijnVisser. LGTM.
